xarray/conventions.py

```diff
@@ -508,6 +528,39 @@ def stackable(dim):
                     new_vars[k].encoding["coordinates"] = coord_str
                     del var_attrs["coordinates"]
                     coord_names.update(var_coord_names)
+            for attr_name in CF_RELATED_DATA:
+                if attr_name in var_attrs:
+                    attr_val = var_attrs[attr_name]
+                    var_names = attr_val.split()
+                    if attr_name in CF_RELATED_DATA_NEEDS_PARSING:
+                        var_names = [
+                            name
+                            for name in var_names
+                            if not name.endswith(":") and not name == k
```

Comment:
This would break if `attrs["cell_measures"] = "area : cell_area volume : cell_volume"`. I don't know if that version is CF-compliant, so maybe we shouldn't worry about it.

PS: cf-xarray does this: https://github.com/xarray-contrib/cf-xarray/blob/5133a4dada23b18221f2136e1f763cda8e65f249/cf_xarray/utils.py#L52-L69

Reply:
The examples do not have a space between the measure and the colon, but it isn't clear if that is allowed.
https://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#cell-measures
https://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#parametric-v-coord

I added a test for that case in `test_backends` (`formula_terms` set to `p0: P0 lev : ln_p` to test both cases at once).